File: codex_poting/java/src/main/java/com/example/demo/data/domain/BookStockMovementType.java
package com.example.demo.data.domain;

import com.fasterxml.jackson.annotation.JsonCreator;
import org.seasar.doma.Domain;

@Domain(valueType = int.class, factoryMethod = "of")
public enum BookStockMovementType {
    INITIAL_STOCK(1),
    PURCHASE(2),
    SALE(3),
    RETURN_PURCHASE(4),
    SALES_RETURN(5),
    STOCK_ADJUSTMENT(6),
    STORE_TRANSFER_IN(7),
    STORE_TRANSFER_OUT(8);

    private final int value;

    BookStockMovementType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    @JsonCreator
    public static BookStockMovementType of(int value) {
        for (BookStockMovementType enm : values()) {
            if (enm.value == value) {
                return enm;
            }
        }
        throw new IllegalArgumentException("Unknown BookStockMovementType value: " + value);
    }
}

File: codex_poting/kotlin/src/main/kotlin/com/example/demo/data/domain/BookStockMovementType.kt
package com.example.demo.data.domain

import com.fasterxml.jackson.annotation.JsonCreator
import org.seasar.doma.Domain

@Domain(valueType = Int::class, factoryMethod = "of")
enum class BookStockMovementType(val value: Int) {
    INITIAL_STOCK(1),
    PURCHASE(2),
    SALE(3),
    RETURN_PURCHASE(4),
    SALES_RETURN(5),
    STOCK_ADJUSTMENT(6),
    STORE_TRANSFER_IN(7),
    STORE_TRANSFER_OUT(8);

    companion object {
        @JvmStatic
        @JsonCreator
        fun of(value: Int): BookStockMovementType {
            for (enm in entries) {
                if (enm.value == value) {
                    return enm
                }
            }
            throw IllegalArgumentException("Unknown BookStockMovementType value: " + value)
        }
    }
}
